hack/verify-release.sh

```diff
@@ -0,0 +1,574 @@
+#!/usr/bin/env bash
+#
+# Copyright 2023 The Metal3 Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# USAGE:
+#
+# This script aims to verify a release content per docs/releasing.md
+# is all done, all images are built and release in general good to go.
+# It can be executed before making a release tag to verify Go dependencies
+# and vulnerabilities are already fixed.
+#
+# Git setup:
+# This script expects to be executed in the root directory of IPAM
+# repository, with the release commit/tag in question checked out.
+#
+# Command line arguments:
+# arg1: mandatory: version without leading v, eg. 1.5.0
+#
+# Environment variables:
+# GITHUB_TOKEN: mandatory: your bearer token that has access to the release
+# REMOTE: optional: use this git remote for tag checks: Default: autodetected
+# CONTAINER_RUNTIME: optional: container runtime binary. Default: docker
+
+set -eu
+# we are using plenty of subshell pipes, and catch errors elsewhere
+set +o pipefail
+
+# enable support for **/go.mod
+shopt -s globstar
+
+# user input
+VERSION="${1:?release version missing, provide without leading v. Example: 1.5.0}"
+GITHUB_TOKEN="${GITHUB_TOKEN:?export GITHUB_TOKEN with permissions to read unpublished release notes}"
+
+# if CONTAINER_RUNTIME is set, we will use crane and osv-scanner from images
+# otherwise, we will expect them to be installed binaries. This allows some
+# flexibility for the Mac users, where Docker Desktop is a bit problematic.
+CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-}"
+# correct remote will be autodetected, if empty
+REMOTE="${REMOTE:-}"
+
+# this repo
+ORG="metal3-io"
+PROJECT="${ORG}/ip-address-manager"
+REGISTRY="quay.io"
+
+# if the given tag doesn't exist, we run only pre-tag checks
+TAG_EXISTS=""
+
+
+#
+# checklist configuration
+#
+
+# git tags
+declare -a git_annotated_tags=(
+    "v${VERSION}"
+)
+
+declare -a git_lightweight_tags=(
+    "api/v${VERSION}"
+    "test/v${VERSION}"
```

Comment:
We don't have test module in IPAM ?

Reply:
True. We had `test` but it is only containing fuzz. Removed.